src/test/scala/com/microsoft/hyperspace/index/DeltaLakeIntegrationTest.scala

```diff
@@ -457,6 +457,74 @@ class DeltaLakeIntegrationTest extends QueryTest with HyperspaceSuite {
     }
   }
 
+  test("Verify time travel query works well with VacuumIndex.") {
+    withTempPathAsString { path =>
+      import spark.implicits._
+      val df = sampleData.toDF("Date", "RGUID", "Query", "imprs", "clicks")
+      df.write.format("delta").save(path)
+
+      val tsMap = mutable.Map[Long, String]()
+      tsMap.put(0, getSparkFormattedTimestamps(System.currentTimeMillis).head)
+
+      val indexName = "deltaIndex3"
+      val deltaDf = spark.read.format("delta").load(path)
+
+      Hyperspace.getContext.indexCollectionManager
+
+      appendAndRefresh(df, path, indexName, None, tsMap) // delta version 1
+      appendAndRefresh(df, path, indexName, None, tsMap) // delta version 2
+
+      withSQLConf(IndexConstants.INDEX_LINEAGE_ENABLED -> "true") {
+        hyperspace.createIndex(deltaDf, IndexConfig(indexName, Seq("clicks"), Seq("Query")))
+      }
+
+      withIndex(indexName) {
+        checkExpectedIndexUsed(indexName, path, None, 1)
+
+        withSQLConf(TestConfig.HybridScanEnabled: _*) {
+          appendAndRefresh(df, path, indexName, None, tsMap)
+          // delta version 3, index log version 1.
+          appendAndRefresh(df, path, indexName, Some("incremental"), tsMap)
+          // delta version 4, index log version 3 (refresh).
+
+          // Without delta table version, the latest log version should be applied.
+          checkExpectedIndexUsed(indexName, path, None, 3)
+          // For delta table version 0, candidate log version is 1.
+          checkExpectedIndexUsed(indexName, path, Some(0, tsMap(1)), 1)
+
+          appendAndRefresh(df, path, indexName, None, tsMap)
+          // delta version 5, index log version 3.
+          appendAndRefresh(df, path, indexName, None, tsMap)
+          // delta version 6, index log version 3.
+          appendAndRefresh(df, path, indexName, Some("incremental"), tsMap)
+          // delta version 7, index log version 5 (refresh).
+          hyperspace.optimizeIndex(indexName)
+          // delta version 7, index log version 7 (optimize).
+          appendAndRefresh(df, path, indexName, Some("incremental"), tsMap)
+          // delta version 8, index log version 9 (refresh).
+          hyperspace.optimizeIndex(indexName)
+          // delta version 8, index log version 11 (optimize).
+          appendAndRefresh(df, path, indexName, None, tsMap)
+          // delta version 9, index long version 11.
+
+          // calling vacuumIndex on active index deletes outdated data and history.
+          hyperspace.vacuumIndex(indexName)
```

Comment:
Could you check if outdated files are removed - using listFiles?

Reply:
Ok. I will.